### PR TITLE
Rewrite then/2 when function is a named function reference

### PR DESCRIPTION
**Pull request by tfiedlerdejanze**
- [x] Please [sign Adobe's CLA](http://opensource.adobe.com/cla.html) if this is your first time contributing to an Adobe open source repo. Thanks!

:wave: first and foremost, thanks for creating and maintaining styler! We use it in a large codebase where i stumbled over some occurrences of named function references passed into `Kernel.then/2`. I couldn't figure when that would be useful so I am opening this PR proposing to have it handled by styler. I've tested this change in our codebase but am happy to get any pointers on what i could've missed!

Let me know if you'd consider adding this to styler (or not! :)

### CODE REVIEW

File: test/style/pipes_test.exs [hunk 1]
```diff
@@ -498,6 +498,13 @@ defmodule Styler.Style.PipesTest do
       assert_style("a |> (fn x -> x end).() |> c", "a |> then(fn x -> x end) |> c()")
     end
 
+    test "rewrites then/2 when the passed function is a named function reference" do
+      assert_style("a |> then(&fun/1) |> c", "a |> fun() |> c()")
+      assert_style("a |> then(&DateTime.from_is8601/1) |> c", "a |> DateTime.from_is8601() |> c()")
+      assert_style("a |> then(&DateTime.from_is8601/1)", "DateTime.from_is8601(a)")
+      assert_style("a |> then(&fun(&1, d)) |> c", "a |> then(&fun(&1, d)) |> c()")
```
Comment (tfiedlerdejanze) [inline test/style/pipes_test.exs:505]: this could actually also be rewritten as `a |> fun(d) |> c()`

**Review by novaugust — APPROVED** [review]
what a fantastic contribution! thanks so much :)
you've timed this well - i just updated our internal credo to disallow `then/2` from the codebase entirely. i'm all for this addition to styler <3

File: test/style/pipes_test.exs (hunk 1 above)
Comment (novaugust) [reply test/style/pipes_test.exs:505]: saw you reverted the commit that _did_ do that - did you see an edge case i'm not? thinking i'll just revert your revert after reimplementing the exact same function 😂

Comment (tfiedlerdejanze) [reply test/style/pipes_test.exs:505]: yeah i am afraid it doesn't handle the following case:

```
"a |> then(&Map.put(&1, :attribute, &1.other_attribute)) |> c()"
```

as it produces the following AST:

```
{{:., [line: 1], [{:__aliases__, [last: [line: 1], line: 1], [:Map]}, :put]},
 [
   {:__block__, [line: 1], [:attribute]},
   {{:., [line: 1], [{:&, [line: 1], [1]}, :other_attribute]},
    [no_parens: true, line: 1], []}
 ]}
 ```

i think it would take another clause or some massaging of the `args` in https://github.com/adobe/elixir-styler/pull/126/commits/59febab7504f6d20f60a450d7717bd83d81e6d02 to make it work properly

Comment (novaugust) [reply test/style/pipes_test.exs:505]: ah thanks! can handle that by verifying that `{&, _, [1]}` doesn't appear anywhere else in the tree

Comment (tfiedlerdejanze) [reply test/style/pipes_test.exs:505]: yes! I am happy to spin up a second PR for this later today

Comment (novaugust) [reply test/style/pipes_test.exs:505]: https://github.com/adobe/elixir-styler/commit/b01d32835c614921129da47844c576784d150923 if you're curious =)

Comment (tfiedlerdejanze) [reply test/style/pipes_test.exs:505]: 🙌 great! Thanks @novaugust :)

Comment (novaugust) [reply test/style/pipes_test.exs:505]: (hah sorry for the poor communication, meant to say "i'll whip that out in a jiffy" but obviously forgot in my enthusiasm to get started)

thanks again for the great idea, you really nailed the spirit of styler ❤️

**Comment by novaugust** [comment]
shipped in 0.11.7 ❤️
